agent/whatweb_agent.py

```diff
@@ -81,39 +81,112 @@ def process(self, message: msg.Message) -> None:
+        targets = []
+        host = message.data.get('host')
+        mask = message.data.get('mask')
+
+        if host is None:
+            return targets
+
+        if mask is not None:
+            try:
+                addresses = ipaddress.ip_network(f'{host}/{mask}')
+                for address in addresses.hosts():
+                    targets.append(Target(name=str(address),
+                                          schema=self._get_schema(message), port=self._get_port(message)))
+            except ValueError as e:
+                logger.error('Invalid IP or mask. %s', e)
         else:
-            raise NotImplementedError(f'Message selector {message.selector} not supported.')
+            try:
+                addresses = ipaddress.ip_network(host)
+                for address in addresses.hosts():
+                    targets.append(Target(name=str(address),
+                                          schema=self._get_schema(message), port=self._get_port(message)))
+            except ValueError as e:
+                logger.error('Invalid IP. %s', e)
 
-        return target
+        return targets
 
-    def _is_target_already_processed(self, message):
+    def _is_target_already_processed(self, message) -> bool:
         """Checks if the target has already been processed before, relies on the redis server."""
-        if message.data.get('url') is not None:
-            unicity_check_key = message.data['url']
-        elif message.data.get('name') is not None:
-            unicity_check_key = message.data['name']
+        if message.data.get('url') is not None or message.data.get('name') is not None:
+            if message.data.get('url') is not None:
+                target = self._get_target_from_url(message.data['url'])
+                unicity_check_key = f'{target.schema}_{target.name}_{target.port}'
+            elif message.data.get('name') is not None:
+                port = self._get_port(message)
+                schema = self._get_schema(message)
+                domain = message.data['name']
+                unicity_check_key = f'{schema}_{domain}_{port}'
 
-        if self.set_add(b'agent_whatweb_asset', unicity_check_key) is True:
-            return True
-        else:
-            logger.info('target %s/ was processed before, exiting', unicity_check_key)
-            return False
+            if self.set_add(b'agent_whatweb_asset', unicity_check_key) is True:
+                return True
+            else:
+                logger.info('target %s/ was processed before, exiting', unicity_check_key)
+                return False
+        elif message.data.get('host') is not None:
+            host = message.data.get('host')
+            mask = message.data.get('mask')
+            if mask is not None:
+                addresses = ipaddress.ip_network(f'{host}/{mask}')
+                result = self.add_ip_network('agent_whois_ip_asset', addresses, lambda net: f'X_{net}_Y')
```

Comment:
The lambda must include schema and port.
lambda net : f'{protocol}_{net}_{protocol}'